### ISSUE TITLE
Incorrect default for ignorecase/smartcase

### ISSUE THREAD

**Issue by Cody157** [VSCodeVim/Vim#642]
### What did you do?

Search using `/` command.
### What did you expect to happen?

Perform case sensitive search
### What happened instead?

Performed case insensitive (+smartcase) search.

I suspect the defaults come from here
https://github.com/VSCodeVim/Vim/blob/master/src/configuration/configuration.ts#L56
But I haven't looked closely enough at the code base to know if that's true.
I personally use case sensitive search (which is the default in vim), but my real concern here is that it's inconsistent with the roadmap
![image](https://cloud.githubusercontent.com/assets/6657951/17901589/aea02bf6-6929-11e6-83f9-dc489cfc4a2c.png)
You can also see the unrelated weirdness of the hlsearch row in that image.
I'm able to override the setting fine with `"vim.ignorecase": false` in the vscode settings, so it's just the default that I believe is wrong.
### Technical details:
- VSCode Version: 1.4.0
- VsCodeVim Version: 0.1.7
- OS: Ubuntu

**Comment by rebornix** [comment]
@Cody157 yup, it's kind of `by design` as both @johnfn and I think `smartcase` is useful and the majority may want to have it turned on so we set the default value our way. Sorry that you don't like it and are you strongly against this kind of setting?

**Comment by Cody157** [comment]
Not at all! I have my preferences, for sure, but I don't see a problem with choosing to make the default what you think is most used. I just believe the roadmap should be consistent with that. Meaning, if it's intentional, I would just make sure that the default is documented correctly.

**Comment by rebornix** [comment]
@Cody157 ha, I see the problem, I still put Vim's default value on the roadmap and that's real misleading. I'll give a turnaround.

**Comment by rebornix** [comment]
@Cody157 thanks for reporting this issue and now the doc should be good. I should put more love in the settings part since it's me who brought it in.

**Comment by johnfn** [comment]
@rebornix if you give me a quick rundown on slack i will write it up :)

**Comment by patricklucas** [comment]
I ran into this issue too. Feel free to respond with "because we said so", but I'm not sure I follow why `vim.ignorecase` defaults to `true` when that's not the case in vim itself

I expect most users of this plugin are coming from vim and used to case-sensitive search, suggesting the default should match vim, but maybe I'm underestimating the number of users of this plugin who haven't actually used vim.

**Comment by J-Fields** [comment]
I think at this point moving away from "sane" defaults to vim's defaults would potentially be a pretty disruptive change, but I also think we should at least make sure we make the fact we're not using vim's defaults clear (especially because the recent .vimrc support means people will be sharing config overrides between VSCode and vim

**Comment by eddy-geek** [comment]
I would like to mention collateral impact on **`*` and `#` (Find next / Find previous)**.

Although I don't strongly care about *ignorecase* default (*smartcase* is certainly friendlier at least to non-vim users), it puts me off that because of it, `*` and `#` will be be case insensitive, **even if the word contains upper-case**. Presumably they are impacted by *ignorecase=True* but not *smartcase=True* -- which is also vim's own behaviour.

Pressing `*` on `CONSTANT = 42` will happily move it to `constant`, which is never what I wants with `*`. Also there are no alternatives, whereas `/`  (1) `/` *smartcase* (2) has \C (3) we have VSCode's `Ctrl+F` and `Alt+C` to toggle case no too far away. But `*` has no such workarounds that I know of (?)

For now I have grudgingly disabled *ignorecase*, but I'd be happy if anyone found a better solution to decorrelate case settings for `/` and `*`

**Comment by J-Fields** [comment]
I agree it's a bit weird but [smartcase does not affect `*`](https://vimhelp.org/options.txt.html#%27smartcase%27)

**Comment by crazyboycjr** [comment]
By setting `vim.ignorecase` and `vim.smartcase` to `false`, I can use vscode's `ctrl-F` or `shift-ctrl-F` to search when I simply wanted to a case-insentitive search. I can also search and replace in a vim-native fashion in vim's command line. Combining both by far I think is pretty ideal.
